src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -516,19 +523,24 @@ private bool CheckIfUnsolicitedFutureHeader(List<BlockHeader> headers)
         }
 
         /// <summary>Sync when handshake is finished.</summary>
-        private async Task OnStateChangedAsync(INetworkPeer peer, NetworkPeerState oldState)
+        private Task OnStateChanged(INetworkPeer peer, NetworkPeerState oldState)
         {
-            if (peer.State == NetworkPeerState.HandShaked)
-                await this.ResyncAsync().ConfigureAwait(false);
+            if (peer.State == NetworkPeerState.HandShaked && this.cachedHeaders.Count == 0)
+                this.RequestHeaders();
+
+            return Task.CompletedTask;
         }
 
         /// <summary>Resets the expected peer tip and last sent tip and triggers synchronization.</summary>
-        public async Task ResetPeerTipInformationAndSyncAsync()
+        public Task ResetPeerTipInformationAndSync()
         {
             this.BestReceivedTip = null;
             this.BestSentHeader = null;
 
-            await this.ResyncAsync().ConfigureAwait(false);
+            if (this.cachedHeaders.Count == 0)
```

Comment:
this should actually clear the `cachedHeaders` and call `RequestHeaders`

Reply:
Oh yes well spotted.